### PR TITLE
some test tweaks

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Moving to not mmapping storage files.
Some tests will have some messy changes.

#### Summary of Changes
Keep the changes simple by preparing some of the changes now in a way that is equivalent. That makes the next test change pr easy to verify as correct.

Fixes #

### CODE REVIEW

File: accounts-db/src/ancient_append_vecs.rs [hunk 1]
```diff
@@ -1665,8 +1666,9 @@ pub mod tests {
                 .stored_accounts
                 .first()
                 .unwrap();
+            let account_shared_data_with_2_refs = account_with_2_refs.to_account_shared_data();
             let pk_with_2_refs = account_with_2_refs.pubkey();
-            let mut account_with_1_ref = account_with_2_refs.to_account_shared_data();
+            let mut account_with_1_ref = account_shared_data_with_2_refs.to_account_shared_data();
```
Comment (brooksprumo) [inline accounts-db/src/ancient_append_vecs.rs]: If `account_shared_data_with_2_refs` is already an AccountSharedData, call we skip calling `.to_account_shared_data()` here?

File: accounts-db/src/ancient_append_vecs.rs [hunk 2]
```diff
@@ -1683,7 +1685,7 @@ pub mod tests {
             append_single_account_with_default_hash(
                 &ignored_storage,
                 pk_with_2_refs,
-                &account_with_2_refs.to_account_shared_data(),
+                &account_shared_data_with_2_refs.to_account_shared_data(),
```
Comment (brooksprumo) [inline accounts-db/src/ancient_append_vecs.rs]: and here

File: accounts-db/src/ancient_append_vecs.rs [hunk 3]
```diff
@@ -1739,7 +1746,7 @@ pub mod tests {
                 vec![&pk_with_1_ref]
             );
             assert_eq!(
-                one_ref_accounts
+                one_ref_accounts_account_shared_data
```
Comment (brooksprumo) [inline accounts-db/src/ancient_append_vecs.rs:1748]: Similar here. There's an call below to `.to_account_shared_data()`, but this `one_ref_accounts_account_shared_data` is already AccountSharedData.

File: accounts-db/src/ancient_append_vecs.rs [hunk 4]
```diff
@@ -1815,7 +1822,7 @@ pub mod tests {
                     .first()
                     .unwrap()
                     .to_account_shared_data(),
-                account_with_2_refs.to_account_shared_data()
+                account_shared_data_with_2_refs.to_account_shared_data()
```
Comment (brooksprumo) [inline accounts-db/src/ancient_append_vecs.rs]: Another one here.

File: accounts-db/src/ancient_append_vecs.rs [hunk 5]
```diff
@@ -1843,8 +1850,9 @@ pub mod tests {
                 .stored_accounts
                 .first()
                 .unwrap();
+            let account_shared_data_with_2_refs = account_with_2_refs.to_account_shared_data();
             let pk_with_2_refs = account_with_2_refs.pubkey();
-            let mut account_with_1_ref = account_with_2_refs.to_account_shared_data();
+            let mut account_with_1_ref = account_shared_data_with_2_refs.to_account_shared_data();
```
Comment (brooksprumo) [inline accounts-db/src/ancient_append_vecs.rs]: Same comments for this test.

File: accounts-db/src/ancient_append_vecs.rs (hunk 1 above)
Comment (jeffwashington) [reply accounts-db/src/ancient_append_vecs.rs]: hopefully all done. please check again

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Pull request merged**